### DISCUSSION TITLE
Not work with ts path aliases

### DISCUSSION THREAD

**Discussion by imcm7** [mikro-orm/mikro-orm#4686] [Q&A]
error TS2307: Cannot find module '~/server/entities/index.ts' or its corresponding type declarations.

Not help `tsconfig-paths/register `
ts-node --project tsconfig.json -r tsconfig-paths/register ./node_modules/@mikro-orm/cli/esm schema:update --run

**Comment by B4nan** [comment]
Can you please provide a complete repro? This is usually a wrong app setup, especially since you are registering the paths mapping explicitly via `-r tsconfig-paths/register`. Note that the ORM does that for the CLI context automatically as long as the paths mapping is specified in the root tsconfig.json:

https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/utils/ConfigurationLoader.ts#L142-L147

> ts-node --project tsconfig.json -r tsconfig-paths/register ./node_modules/@mikro-orm/cli/esm schema:update --run

This seems wrong, you cannot use `ts-node` to run ESM project, you need to use `ts-node-esm` instead, maybe that will help? Also afaik this is broken in node 20, see https://github.com/TypeStrong/ts-node/issues/1997#issuecomment-1537111441

**Comment by imcm7** [comment]
@B4nan https://github.com/imcm7/nuxt-mikroorm-ts

`pnpm db:update`

`Error: Cannot find module '~/server/entities'`

**Reply by imcm7** [reply]
`node -v v20.5.0`
But it not work with `v19.9.0` and `v18` too.
And when I usend 18 I think too.

And I think the auto import of entities also not work, in past i used entities/index.ts for that esbuild
